mempool/reactor.go

```diff
@@ -257,21 +253,18 @@ func (memR *Reactor) broadcastTxRoutine(peer p2p.Peer) {
 //-----------------------------------------------------------------------------
 // Messages
 
-// Message is a message sent or received by the Reactor.
-type Message interface{}
-
-func RegisterMessages(cdc *amino.Codec) {
-	cdc.RegisterInterface((*Message)(nil), nil)
-	cdc.RegisterConcrete(&TxMessage{}, "tendermint/mempool/TxMessage", nil)
-}
-
-func (memR *Reactor) decodeMsg(bz []byte) (msg Message, err error) {
+func (memR *Reactor) decodeMsg(bz []byte) (TxMessage, error) {
 	maxMsgSize := calcMaxMsgSize(memR.config.MaxTxBytes)
 	if l := len(bz); l > maxMsgSize {
-		return msg, ErrTxTooLarge{maxMsgSize, l}
+		return TxMessage{}, ErrTxTooLarge{maxMsgSize, l}
+	}
+
+	msg := gogotypes.BytesValue{}
+	err := msg.Unmarshal(bz)
+	txMsg := TxMessage{
+		Tx: types.Tx(msg.Value),
```

Comment:
Looks like `gogotypes.BytesValue` and `TxMessage` are basically equivalent. Only difference is the field name. I'm wondering if this cast/conversion to `Tx` copies the value and if yes, if this is desired.

Reply:
Conversion doesn't do copies apparently (only changes the type not the represntation), so this shouldn't be a concern: https://golang.org/ref/spec#Conversions